core/db_classes/EE_Registration.class.php

```diff
@@ -160,6 +160,9 @@ public function set_status($new_STS_ID = null, $use_default = false, ContextInte
             && ! empty($new_STS_ID) // as well as the new status
             && $this->ID() // ensure registration is in the db
         ) {
+            // update internal status first
+            parent::set('STS_ID', $new_STS_ID, $use_default);
```

Comment:
I agree, this looks like the way things _should_ have been originally.  However, this does break back compat for the existing actions in this method that pass along this instance because prior to this change, the actions would have been fired _before_ the internal status had changed on the instance but after this change the actions are fired _after_ the internal status change.  Granted, we'd expect that client code would be utilizing the status variables passed along to the action callback but it's still a potential gotcha that I thought worth mentioning.

Does it mean we don't do this fix?  No.  But I think it may be worth emphasizing in any changelog notes (maybe even a brief developers.eventespresso.com blog post).  As well, I'm assuming you've done a search through all our official code (core and add-ons) to see if anything hooking into these actions would be affected.

Reply:
Both Michael and I searched our local codebases and could not find ANY uses of either of those two hooks so it's unlikely that others are using them if we are not doing so ourselves. That said, I can make add a brief post to http://developer.eventespresso.com to give a heads up. I don't think the changelog entry should stray too far from convention because we don't want to make a precedent out of writing overly verbose entries.